Makefile

```diff
@@ -382,7 +384,7 @@ endif
 WARNINGS = all error strict-prototypes missing-prototypes nested-externs \
 	missing-declarations undef pointer-arith no-nonnull declaration-after-statement
 
-CFLAGS += --std=c99 -nostdlib -nostdinc -ffreestanding \
+CFLAGS += --std=c99 -nostdlib -nostdinc -ffreestanding -fms-extensions \
```

Comment:
in file include/api/debug.h
```
typedef struct {
    debug_entry_type_t path;
    union {
        struct {
            word_t irq;
        };
        struct {
            word_t word;
        };
        struct {
            word_t number;
            word_t code;
        };
        struct {
            word_t fault_type;
        };
        struct {
            word_t syscall_no;
            word_t cap_type;
            word_t invocation_tag;
        };
    };
} debug_entry_t;
```
on my old gcc will got error
```
./include/api/debug.h:34:10: error: declaration does not declare anything
./include/api/debug.h:37:10: error: declaration does not declare anything
./include/api/debug.h:41:10: error: declaration does not declare anything
./include/api/debug.h:44:10: error: declaration does not declare anything
./include/api/debug.h:49:10: error: declaration does not declare anything
./include/api/debug.h:50:6: error: declaration does not declare anything
```

this is c11 feature, explain
https://gcc.gnu.org/onlinedocs/gcc/Unnamed-Fields.html